lib/ansible/modules/cloud/google/gce_snapshot.py

```diff
@@ -0,0 +1,195 @@
+- name: Create snapshots of specific disks
+  gce_snapshot:
+    instance_name: example-instance
+    snapshot_name: example-snapshot
+    state: present
+    disks:
+      - disk0
+      - disk1
+    service_account_email: project_name@appspot.gserviceaccount.com
+    credentials_file: /path/to/credentials
+    project_id: project_name
+  delegate_to: localhost
+'''
+
+try:
+    from libcloud.compute.types import Provider
+    _ = Provider.GCE
+    HAS_LIBCLOUD = True
+except ImportError:
+    HAS_LIBCLOUD = False
+
+
+def find_snapshot(volume, name):
+    '''
+    Check if there is a snapshot already created with the given name for
+    the passed in volume.
+
+    Args:
+        volume: A gce StorageVolume object to manage
+        name: The name of the snapshot to look for
+
+    Returns:
+        The VolumeSnapshot object if one is found
+    '''
+    found_snapshot = None
+    snapshots = volume.list_snapshots()
+    for snapshot in snapshots:
+        if name == snapshot.name:
+            found_snapshot = snapshot
+    return found_snapshot
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            instance_name=dict(required=True),
+            snapshot_name=dict(required=True),
+            state=dict(choices=['present', 'absent'], default='present'),
+            disks=dict(default=None, type='list'),
+            service_account_email=dict(type='str'),
+            credentials_file=dict(type='path'),
+            project_id=dict(type='str')
+        )
+    )
+
+    if not HAS_LIBCLOUD:
+        module.fail_json(msg='libcloud with GCE support (0.19.0+) is required for this module')
+
+    gce = gce_connect(module)
+
+    instance_name = module.params.get('instance_name')
+    snapshot_name = module.params.get('snapshot_name')
+    disks = module.params.get('disks')
+    state = module.params.get('state')
+
+    changed = False
+    snapshot = None
+    msg = ''
+
+    instance = gce.ex_get_node(instance_name, 'all')
+    instance_disks = instance.extra['disks']
+
+    for instance_disk in instance_disks:
+        device_name = instance_disk['deviceName']
+        if disks is None or device_name in disks:
+            volume_obj = gce.ex_get_volume(device_name)
+
+            # If we have more than one disk to snapshot, prepend the disk name
+            if len(instance_disks) > 1:
+                snapshot_name = device_name + "-" + snapshot_name
+
+            snapshot = find_snapshot(volume_obj, snapshot_name)
+
+            if snapshot and state == 'present':
+                msg = snapshot_name + " already exists"
+
+            elif snapshot and state == 'absent':
+                snapshot.destroy()
+                changed = True
+                msg = snapshot_name + " was deleted"
+
+            elif not snapshot and state == 'present':
+                volume_obj.snapshot(snapshot_name)
+                changed = True
+                msg = snapshot_name + " created"
```

Comment:
As this is in a loop, the message will be overwritten on each iteration, with only the last snapshot created being reported.